unittest/progress_test.cc

```diff
@@ -0,0 +1,161 @@
+///////////////////////////////////////////////////////////////////////
+// File:        apiexample_test.cc
+// Description: Progress reporting API Test for Tesseract.
+// Author:      Jaroslaw Kubik
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+///////////////////////////////////////////////////////////////////////
+
+// expects clone of tessdata_fast repo in ../../tessdata_fast
+
+#include "include_gunit.h"
+#include "gmock/gmock.h"
+#include "baseapi.h"
+#include "ocrclass.h"
+#include "leptonica/allheaders.h"
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <locale>
+#include <limits.h>
+#include <time.h>
+
+namespace {
+
+class QuickTest : public testing::Test {
+ protected:
+  virtual void SetUp() {
+    start_time_ = time(nullptr);
+  }
+  virtual void TearDown() {
+    const time_t end_time = time(nullptr);
+    EXPECT_TRUE(end_time - start_time_ <=25) << "The test took too long - " << ::testing::PrintToString(end_time - start_time_);
+  }
+  time_t start_time_;
+  };
+
+  class ClassicMockProgressSink {
+  public:
+    MOCK_METHOD1(classicProgress, bool( int ) );
+    MOCK_METHOD1(cancel, bool( int ));
+
+    ETEXT_DESC monitor;
+
+    ClassicMockProgressSink()
+    {
+        monitor.progress_callback = []( int progress, int, int, int, int ) ->bool {
+          return instance->classicProgress( progress );
+        };
+        monitor.cancel = []( void* ths, int words ) -> bool {
+          ((ClassicMockProgressSink*)ths)->cancel( words );
+        };
```

Comment:
Here the compiler issues a warning:

    ../../../../unittest/progress_test.cc: In Lambda-Funktion:
    ../../../../unittest/progress_test.cc:59:9: Warnung: keine return-Anweisung in nicht void zurückgebender Funktion [-Wreturn-type]
             };
             ^

It is also reported by Coverity Scan ("CID 1393240 Missing return statement").